### PR TITLE
feat: added UI for Wave Generator Activity

### LINKED ISSUE #923: Create the monitors for Wave Generator
**Actual Behaviour**

There are no monitors in wave generator activity.

**Expected Behaviour**

wave generator should have monitors to view the changes while controlling waveforms.

**Steps to reproduce it**

Create WaveGenerator Activity -> Add Monitors.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Avjeet**
Fixes #923 
First steps for issue #878 

Changes: Created WaveGenerator Activity Class and added two monitors showing relative position and color of all the text and icon that has to be shown on the monitors.

Screenshots for the change: 
image

APK for testing:
I have provided the apk for testing screen on different layouts.
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2041991/app-debug.zip)

### CODE REVIEW

**Comment by Avjeet** [comment]
@harsh-2711 @abhinavraj23 @CloudyPadmal @Vikum94 plz test the screen on different phone screen sizes.

File: app/src/main/res/drawable/custom_seekbar.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (CloudyPadmal) [inline app/src/main/res/drawable/custom_seekbar.xml]: What if we used the existing seekbar without inventing a new one?

**Comment by CloudyPadmal** [comment]
We can have the frequency value in a bigger font and a red color with a black background is a little hard to view.

**Comment by Vikum94** [comment]
May be you can use Orange or green as in your first mockup instead of red text

File: app/src/main/res/drawable/custom_seekbar.xml (hunk 1 above)
Comment (Avjeet) [reply app/src/main/res/drawable/custom_seekbar.xml]: I made this seekbar because the existing seek bar doesnt look good with whole screen , it only suits better with smaller card views.

**Comment by Avjeet** [comment]
image
changed the color and increased the font size.

**Comment by harsh-2711** [comment]
@Avjeet Please try to use the design of the original seekbar as this seekbar doesn't look good. And I can't figure out why you left that empty space below. Also the frrequency should be mentioned side by side with the heading and not below it. And don't use bold texts for buttons

**Comment by Avjeet** [comment]
@harsh-2711 this PR is only for the monitors I will create rest of the functionality in subsequent PRs

**Comment by CloudyPadmal** [comment]
@Avjeet please have a look at [this](https://github.com/warkiz/IndicatorSeekBar) library for seekbars and implement a _discrete_ seekbar between the frequency range we're supporting.

**Comment by Avjeet** [comment]
updated the layout with discrete seek bar
image

**Comment by Avjeet** [comment]
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2042834/app-debug.zip)
@CloudyPadmal the new apk with discrete seek bar

**Comment by harsh-2711** [comment]
@Avjeet  This looks good..

File: app/build.gradle [hunk 2]
```diff
@@ -53,6 +54,7 @@ dependencies {
     testImplementation 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.4'
 
     compile 'com.github.mirrajabi:search-dialog:1.1'
+    compile 'com.github.warkiz.widget:indicatorseekbar:1.2.9'
```
Comment (CloudyPadmal) [inline app/build.gradle]: Change `compile` to `implementation`. `compile` in `56` is already changed.

**Comment by CloudyPadmal** [comment]
@Avjeet how's the progress with this?

**Comment by CloudyPadmal** [comment]
@Avjeet please follow up on this issue quickly as we are at the edge of [Milestone - 1](https://github.com/fossasia/pslab-android/milestone/17)

**Comment by Avjeet** [comment]
@CloudyPadmal @mariobehling @Vikum94 I have completed the UI for Wave Generator. This layout will show button animations for devices with api>=21 and it uses constraint layout so it will stretch properly on different screen sizes.
image

**Comment by Avjeet** [comment]
@CloudyPadmal @Vikum94 @mariobehling Added the complete layout of Wave Generator Activity as it uses constraint layout it will stretch to different screen sizes.
image

**Comment by CloudyPadmal** [comment]
@Avjeet this is nice. Can you provide the APK to test the layout? Also add the resources in resource files to the middle as I've mentioned in the public chat yesterday to avoid unnecessary merge conflicts in every other PR

File: app/src/main/res/values/colors.xml [hunk 3]
```diff
@@ -11,4 +11,9 @@
     <color name="customBorderFill">#7A7A7A</color>
     <color name="blue_grey">#546E7A</color>
     <color name="grey_light">#bdbdbd</color>
+    <color name="grey">#2D2F31</color>
+    <color name="white_grey">#9ea7aa</color>
+    <color name="dark_grey">#757575</color>
+    <color name="orange">#ff7043</color>
+>>>>>>> Added monitors for wavegen activity
```
Comment (CloudyPadmal) [inline app/src/main/res/values/colors.xml]: Remove this line and add the new colors may be before `blue_grey`?

File: app/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -133,4 +133,31 @@
...
+    <dimen name="ctrl_panel_margin_bottom">5dp</dimen>
+    <dimen name="ctrl_sepration_margin">10dp</dimen>
+    <dimen name="back_margin_top">10dp</dimen>
+    <dimen name="ctrl_inside_rl_padding">5dp</dimen>
+    <dimen name="ctrl_title_text_margin">3dp</dimen>
+    <dimen name="ctrl_title_text_padding">5dp</dimen>
+    <dimen name="view_btn_margin">10dp</dimen>
+    <dimen name="mon_title_sep_margin">16dp</dimen>
+    <dimen name="length_0dp">0dp</dimen>
+    <dimen name="divider_width">1dp</dimen>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/dimens.xml]: Same with this. Add these new dimens before `device_version_padding`

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -669,4 +670,33 @@
...
+    <string name="unit_percent">%</string>
+    <string name="text_freq">Freq</string>
+    <string name="text_waveform">Waveform</string>
+    <string name="text_pwm">PWM</string>
+    <string name="view_text">View</string>
+    <string name="phase_holder_text">45 deg</string>
+    <string name="duty_holder_text">50 %</string>
+
+
+
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: Put the new strings before `text_connected`

**Comment by Avjeet** [comment]
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2065792/app-debug.zip)
@CloudyPadmal @Vikum94 changes done and here is the apk.

File: app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -57,7 +59,14 @@
...
 
-        <activity android:name=".activity.PowerSourceActivity"></activity>
+
+        <activity android:name=".activity.PowerSourceActivity"/>
+
+        <activity android:name=".activity.WaveGeneratorActivity"
+            android:screenOrientation="userLandscape">
+
+        </activity>
+
```
Comment (CloudyPadmal) [inline app/src/main/AndroidManifest.xml]: Put `activities` together. Before `receiver`

File: app/src/main/res/drawable/btn_back_rounded.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<shape android:shape="rectangle"
+    xmlns:android="http://schemas.android.com/apk/res/android">
+
+            <solid android:color="@color/colorPrimary" />
+            <corners android:radius="5dp" />
+        </shape>
```
Comment (CloudyPadmal) [inline app/src/main/res/drawable/btn_back_rounded.xml]: This file needs a reformat

File: app/src/main/res/drawable/rectangle_border.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <corners android:radius="10dp"/>
+    <stroke android:color="@color/colorPrimary"
+        android:width="3dp"/>
+
```
Comment (CloudyPadmal) [inline app/src/main/res/drawable/rectangle_border.xml]: Extra line

File: app/src/main/res/drawable/btn_back_rounded.xml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<shape android:shape="rectangle"
+    xmlns:android="http://schemas.android.com/apk/res/android">
+
```
Comment (CloudyPadmal) [inline app/src/main/res/drawable/btn_back_rounded.xml]: Extra line

File: app/src/main/res/values/dimens.xml [hunk 10]
```diff
@@ -126,11 +153,11 @@
...
+    <dimen name="separator_width">1dp</dimen>
     <dimen name="device_status_margin_bottom">40dp</dimen>
     <dimen name="connect_inst_margin_b0ttom">20dp</dimen>
     <dimen name="device_status_height_text">60dp</dimen>
     <dimen name="progress_bar_height">8dp</dimen>
     <dimen name="connect_instruction_text_header">18sp</dimen>
 
-</resources>
\ No newline at end of file
+</resources>
```
Comment (CloudyPadmal) [inline app/src/main/res/values/dimens.xml]: Revert these changes at the end of file

**Comment by Avjeet** [comment]
@CloudyPadmal all minor changes done 👍

File: app/build.gradle [hunk 11]
```diff
@@ -50,6 +50,8 @@ dependencies {
     implementation "com.github.tiagohm.MarkdownView:library:$rootProject.markDownViewVersion"
     implementation "com.github.mirrajabi:search-dialog:$rootProject.searchDialogVersion"
     implementation "com.github.BeppiMenozzi:Knob:$rootProject.knobLibraryVersion"
+    implementation 'com.github.warkiz.widget:indicatorseekbar:1.2.9'
```
Comment (CloudyPadmal) [inline app/build.gradle]: @Avjeet please add the version to config file like the rest of the libraries

**Comment by Vikum94** [comment]
Next time onward you can update the apk at the 'apk for testing' in pr message

**Comment by Avjeet** [comment]
@CloudyPadmal If everything is fine can u merge the PR.
